azure/types.go

```diff
@@ -358,8 +359,30 @@ type ManagedClusterSpec struct {
 
 // AgentPoolSpec contains agent pool specification details.
 type AgentPoolSpec struct {
-	Name         string
-	SKU          string
-	Replicas     int32
+	// Name is the name of agent pool.
+	Name string
+
+	// ResourceGroup is the name of the Azure resource group for the AKS Cluster.
+	ResourceGroup string
+
+	// Cluster is the name of the AKS cluster.
+	Cluster string
+
+	// Version defines the desired Kubernetes version.
+	Version *string
+
+	// SKU defines Azure VM size for the agent pool VMs.
+	SKU string
+
+	// Replicas is the number of desired machines.
+	Replicas int32
+
+	// OSDiskSizeGB is OS disk size in GB for every machine in this agent pool.
 	OSDiskSizeGB int32
+
+	// VnetSubnetID is the Azure Resource ID for the subnet which should contain nodes.
+	VnetSubnetID string
+
+	// Mode represents mode of an agent pool. Possible values include: 'AgentPoolModeSystem', 'AgentPoolModeUser'.
```

Comment:
these are the Azure values, not what we expose right? I guess doc'ing them here is fine, but they shouldn't be exposed really.

Reply:
Oh right, thanks. I saw that we just use the same values like Azure SDK code, so I copied this from [code docs from Azure SDK](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-05-01/containerservice@v55.2.0+incompatible#ManagedClusterAgentPoolProfileProperties.Mode) (which are then not really correct strings, those are const names from the code :facepalm:).

Reply:
Fixed.

Reply:
The values that are in the code doc now are the same as in `AzureManagedMachinePoolSpec.Mode`, since those are used here as well.